File: app/src/main/java/com/hmproductions/bingo/dagger/PreferencesModule.java
package com.hmproductions.bingo.dagger;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import dagger.Module;
import dagger.Provides;

@Module
public class PreferencesModule {

    @Provides
    @BingoApplicationScope
    public SharedPreferences getSharedPreferences(Context context) {
        return PreferenceManager.getDefaultSharedPreferences(context);
    }
}

File: app/src/main/java/com/hmproductions/bingo/dagger/PreferencesModule.kt
package com.hmproductions.bingo.dagger

import android.content.Context
import android.content.SharedPreferences
import android.preference.PreferenceManager
import dagger.Module
import dagger.Provides

@Module
class PreferencesModule {

    @Provides
    @BingoApplicationScope
    fun getSharedPreferences(context: Context): SharedPreferences = PreferenceManager.getDefaultSharedPreferences(context)
}
